header/dfr_syncfree.h

```diff
@@ -0,0 +1,30 @@
+#define WARP_PER_BLOCK 28
+#define WARP_SIZE 32
+
+typedef struct {
+	int nr; //Number of rows
```

Comment:
If you have to comment what the variable is, then its a clear sign you need to make them more descriptive!